src/Database/Expression/WindowInterface.php

```diff
@@ -0,0 +1,148 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Expression;
+
+/**
+ * This defines the functions used for building window expressions.
+ */
+interface WindowInterface
+{
+    /**
+     * 'CURRENT ROW' frame start, end or exclusion
+     *
+     * @var int
+     */
+    public const CURRENT_ROW = 0;
+
+    /**
+     * 'UNBOUNDED PRECEDING' and '(offset) PRECEDING' frame start or end
```

Comment:
These can only work for the array format otherwise how can we tell the difference between offset integers and constants.  Users will have to specify `null` to mean unbounded.